[13.0][MIG] module base_optional_quick_create

kimirondon opened the pull request.

eLBati committed:
base_optional_quick_create

sbidoul committed:
move all modules to __unported__ on master branch and set installable=False

Commit:
Port to v8 PEP-8 compliant

lmignon committed:
This fixes #73 Check if the model still exists in the registry before registring the 'check_quick_create' hook

gurneyalex committed:
Add OCA as author of OCA addons

In order to get visibility on https://www.odoo.com/apps the OCA board has
decided to add the OCA as author of all the addons maintained as part of the
association.

yvaucher committed:
Add missing default oca icons

sbidoul committed:
prefix versions with 8.0

pedrobaeza committed:
Make modules uninstallable

Commit:
from 8.0 to 9.0 base_optional_quick_create * bump version to 9.0.1.0.0 and make it installable * apply a views/models directory arch into the module * fix xpath anchor on ir_model views (osv_memory to transient) * add README.rst * apply some new api

Commit:
introduce the same behavior than 8.0

sbidoul committed:
base_optional_quick_create: fix bug detected by pylint

pedrobaeza committed:
Make modules uninstallable

pedrobaeza committed:
Rename manifest files

alexis-via committed:
Migrate base_optional_quick_create to Odoo 10

alexis-via committed:
FIX when have a "dead" model that is still in DB

SimoRubi committed:
base_optional_quick_create clears _api field of name_create method, unsetting the flag clears the patch

Currently translated at 100,0% (3 of 3 strings)

Translation: server-tools-10.0/server-tools-10.0-base_optional_quick_create
Translate-URL: https://translation.odoo-community.org/projects/server-tools-10-0/server-tools-10-0-base_optional_quick_create/de/

Commit:
[12.0][MIG] base_optional_quick_create

alexis-via committed:
base_optional_quick_create: avoid a crash during migration (up-port from v10)

oca-travis committed:
[UPD] Update base_optional_quick_create.pot

OCA-git-bot committed:
[UPD] README.rst

llacroix commented on base_optional_quick_create/__manifest__.py:
```diff
@@ -0,0 +1,19 @@
+
+{
+    'name': 'Optional quick create',
+    'version': '12.0.1.0.0',
```
Update version to 13

llacroix commented on base_optional_quick_create/i18n/base_optional_quick_create.pot:
```diff
@@ -0,0 +1,32 @@
+#
+msgid ""
+msgstr ""
+"Project-Id-Version: Odoo Server 12.0\n"
```
Re-export translations in POT for 13.0?

llacroix commented on base_optional_quick_create/readme/CONTRIBUTORS.rst:
```diff
@@ -0,0 +1,4 @@
+* Jonathan Nemry jonathan.nemry@acsone.eu
+* Lorenzo Battistini lorenzo.battistini@agilebg.com
+* Alexis de Lattre alexis.delattre@akretion.com
+* Gilles Meyomesse gilles.meyomesse@acsone.eu (https://acsone.eu)
```
Add yourself as contributor

llacroix reviewed (commented):
Rebuild the README.rst file the oca-maintainer tools 

https://github.com/OCA/maintainer-tools

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-ux-12.0/server-ux-12.0-base_optional_quick_create
Translate-URL: https://translation.odoo-community.org/projects/server-ux-12-0/server-ux-12-0-base_optional_quick_create/

AaronHForgeFlow commented:
It seems commit history is missing. If you have any doubt, just follow the steps here: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-13.0#technical-method-to-migrate-a-module-from-120-to-130-branch

if you need assistance, just ask for it.

Regards.

AaronHForgeFlow reviewed (changes requested):
I tested by making product.product model to "Avoid Quick Create". But I can still create products from purchase order lines, for example. It seems not working.

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-ux-12.0/server-ux-12.0-base_optional_quick_create
Translate-URL: https://translation.odoo-community.org/projects/server-ux-12-0/server-ux-12-0-base_optional_quick_create/

kimirondon commented:
> It seems commit history is missing. If you have any doubt, just follow the steps here: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-13.0#technical-method-to-migrate-a-module-from-120-to-130-branch
> 
> if you need assistance, just ask for it.
> 
> Regards.

hello!
Sorry i couldn't answer you before!! thanks a lot for the info.
Regards

kimirondon commented:
> I tested by making product.product model to "Avoid Quick Create". But I can still create products from purchase order lines, for example. It seems not working.

the right behavior should be raise the product form view when the user tries to create a product from the purchase order lines for example. I will check.
Regards

mcassuto commented:
@kimirondon any update ?

hitrosol commented:
I have tried this locally for models : product.product, res.partner and account.account.
It works well 👍

etobella reviewed (changes requested):
@kimirondon The commit history has not been maintained, can you review it please????

kimirondon commented:
@etobella yes i will !

kimirondon commented:
@etobella i recovered the commit history, let me know if it's ok!

etobella commented:
It seems right!! Thanks :smile:

mcassuto commented:
we have this one on production on multiple instances => could we have it merged ?

AaronHForgeFlow commented:
Still not working on runbot:
![image](https://user-images.githubusercontent.com/19620251/114694345-ba957680-9d1a-11eb-8a96-e0e51ea97456.png)

![image](https://user-images.githubusercontent.com/19620251/114694233-9e91d500-9d1a-11eb-9f23-13fd5555deac.png)

mcassuto commented:
@kimirondon where is the mismatch ?

mcassuto commented:
@AaronHForgeFlow hello Aaaon, I would like your feedback on the MR for v14.

AaronHForgeFlow reviewed (approved):
Ok I see the purpose of the module now. It works. Thanks @mcassuto

mcassuto commented:
@AaronHForgeFlow let's be honnest...
This module works as it and somehow answer the need; it has been approved by customers as well (v13 and v14)...
....
However, I guess you were saying it was NOT working because, like me, you were used to the previous versions where these 2 options (create / create & edit) were simply removed.

I don't know which version did see this change in the behaviour but it has been approved and merged....
@kimirondon did simply migrate the v12 version.

So, what I suggest : 
1/ we merge v13 and v14 now (it could be helpful to a lot of users)
2/ we submit later another version (with a system parameter to enforce the restriction ??)

What do you think ?

AaronHForgeFlow commented:
I think we can continue with this module as it is. 

I was confused with this other module: https://github.com/OCA/web/tree/13.0/web_m2x_options This one actually removes the " (create / create & edit)" options but not at model level.

mcassuto commented:
@AaronHForgeFlow thanks ! then what is the next step to have these 2 modules merged ?

AaronHForgeFlow commented:
Get more approvals including the approval of maintainer of this repository, I am not maintainer of this one.

I see that @etobella  requested changes here regarding the commit history and I am not sure it is fixed. It is kind of weird, the oldest commit and the newest are the same commit?  

First time I see this.

llacroix commented:
@pedrobaeza any idea who could get that verified to be merged?

pedrobaeza commented:
Check on the module contributors for pinging them, but you can add your review as well.

llacroix commented:
@LoisRForgeFlow could you have a look to this PR?

LoisRForgeFlow reviewed (changes requested):
The commit history appear to not be clean, it has migration commits at the beggining and end of history:

![image](https://user-images.githubusercontent.com/23449160/126604537-64c58ae2-c8c8-4768-8cb6-1e245b591913.png)

![image](https://user-images.githubusercontent.com/23449160/126604558-ed4f5e3a-cfac-44df-87bd-8f75c4e851d8.png)

llacroix commented:
@LoisRForgeFlow anything ?

simahawk commented:
@etobella ping

llacroix commented:
anytime would be great

kimirondon commented:
> The commit history appear to not be clean, it has migration commits at the beggining and end of history:
> 
> ![image](https://user-images.githubusercontent.com/23449160/126604537-64c58ae2-c8c8-4768-8cb6-1e245b591913.png)
> 
> ![image](https://user-images.githubusercontent.com/23449160/126604558-ed4f5e3a-cfac-44df-87bd-8f75c4e851d8.png)

Commit:
[13.0][MIG] base_optional_quick_create: Migration to 13.0

Merge the changes again

Format the code

Fix end of files

Remove multi

Add unit test for a new model

Add setup
